File: packages/ios/ApiContracts/Sources/ApiContracts/Models/StatsModels.swift
import Foundation

// MARK: - Stats

public struct SubjectBreakdownItem: Codable, Sendable {
    public let correct: Int
    public let wrong: Int
    public let avgScore: Int
}

public struct RecentActivityItem: Codable, Sendable, Identifiable {
    public let date: String
    public let count: Int

    public var id: String { date }
}

public struct StatsResponse: Codable, Sendable {
    public let totalQuestions: Int
    public let totalWrong: Int
    public let accuracy: Int
    public let avgScore: Int
    public let subjectBreakdown: [String: SubjectBreakdownItem]
    public let recentActivity: [RecentActivityItem]
    public let trend: [StatsTrendItem]?
    public let mastery: [MasteryStatsItem]?
    public let abilities: [String: Double]?
    public let abilityTrend: [AbilityTrendItem]?
}

public struct StatsTrendItem: Codable, Sendable, Identifiable {
    public let date: String
    public let count: Int
    public let accuracy: Int
    public let avgScore: Int
    public var id: String { date }
}

public struct MasteryStatsItem: Codable, Sendable, Identifiable {
    public let knowledgePoint: String
    public let subject: String
    public let level: Double
    public let trend: String
    public let lastSeen: String
    public var id: String { "\(subject)-\(knowledgePoint)" }
}

public struct AbilityTrendItem: Codable, Sendable, Identifiable {
    public let date: String
    public let abilities: [String: Double]
    public var id: String { date }
}

public struct BankCountResponse: Codable, Sendable {
    public let count: Int
}

File: packages/android/apicontracts/src/main/kotlin/com/aistudy/apicontracts/StatsModels.kt
package com.aistudy.apicontracts

import kotlinx.serialization.Serializable

// MARK: - Stats
// GET /api/stats → 学习统计仪表盘

@Serializable
data class StatsResponse(
    val totalQuestions: Int,
    val totalWrong: Int,
    val accuracy: Double,
    val avgScore: Double,
    val subjectBreakdown: Map<String, SubjectBreakdownItem> = emptyMap(),
    val recentActivity: List<RecentActivityItem> = emptyList(),
)

@Serializable
data class SubjectBreakdownItem(
    val correct: Int,
    val wrong: Int,
    val avgScore: Double,
)

@Serializable
data class RecentActivityItem(
    val date: String,
    val count: Int,
)

// MARK: - Bank Count (无鉴权)
// GET /api/bank/count → { count }

@Serializable
data class BankCountResponse(
    val count: Int,
)
